packer/rpc/communicator.go

```diff
@@ -97,7 +112,9 @@ func (c *communicator) Start(cmd *packer.RemoteCmd) (err error) {
 
 		var finished CommandFinished
 		decoder := gob.NewDecoder(conn)
-		if err := decoder.Decode(&finished); err != nil {
+		err = decoder.Decode(&finished)
+		wg.Wait()
```

Comment:
Do we lose error output if we wait after the `if err != nil` block?

If these are functionally the same I'd put the `Wait` after for readability.